### PR TITLE
Document azure ops.

### PR DESCRIPTION
**Pull request by RandySheriffH**
- Document azure ops;
- Set build/src path for triton to solve windows local build failures;
- Create virtual function to allow each invoker set their own full-auth-token;

### CODE REVIEW

File: cmake/externals/triton.cmake [hunk 1]
```diff
@@ -128,6 +128,8 @@ ExternalProject_Add(triton
                     URL https://github.com/triton-inference-server/client/archive/refs/heads/${triton_VERSION_TAG}.tar.gz
                     URL_HASH SHA1=b8fd2a4e09eae39c33cd04cfa9ec934e39d9afc1
                     PREFIX ${triton_PREFIX}
+                    SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/_deps/triton-src
```
Comment (RandySheriffH) [inline cmake/externals/triton.cmake:131]: This change solves my local error - "cl.exe cannot compile a test program."

File: operators/azure/azure_invokers.cc [hunk 2]
```diff
@@ -92,4 +92,8 @@ void AzureTextToTextInvoker::ProcessResponse(const std::string& response, ortc::
   string_tensor.SetStringOutput(std::vector<std::string>{response}, std::vector<int64_t>{1});
 }
 
+std::string AzureTextToTextInvoker::ComposeFullAuthToken(const std::string& auth_token) const {
+  return std::string{"api-key: "} + auth_token;
```
Comment (RandySheriffH) [inline operators/azure/azure_invokers.cc:96]: This is required by azure chat endpoint.
There is also an UT added for the endpoint in test_azure_ops.py.

File: docs/custom_ops.md [hunk 3]
```diff
@@ -1230,6 +1230,349 @@ expect(node, inputs=[value, mask], outputs=[output],
 ```
 </details>
 
+
```
Comment (wenbingl) [inline docs/custom_ops.md:1233]:
```suggestion
## Azure endpoint Operators

```

File: test/data/azure/create_basic_models.py [hunk 4]
```diff
@@ -33,23 +33,23 @@ def create_audio_model():
...
+def create_azure_chat_model():
     auth_token = helper.make_tensor_value_info('auth_token', TensorProto.STRING, [-1])
     chat = helper.make_tensor_value_info('chat', TensorProto.STRING, [-1])
     response = helper.make_tensor_value_info('response', TensorProto.STRING, [-1])
 
     invoker = helper.make_node('AzureTextToText', ['auth_token', 'chat'], ['response'],
                                domain='com.microsoft.extensions',
                                name='chat_invoker',
-                               model_uri='https://api.openai.com/v1/chat/completions',
+                               model_uri='https://rashuai-aoai-test.openai.azure.com/openai/deployments/randysgpt/chat/completions?api-version=2023-05-15',
```
Comment (wenbingl) [inline test/data/azure/create_basic_models.py:44]: is this endpoint temporary or forever?

File: test/test_azure_ops.py [hunk 5]
```diff
@@ -118,18 +118,18 @@ def test_open_ai_audio(self):
...
-    def test_open_ai_chat(self):
+    def test_azure_chat(self):
         if self.__enabled:
-            sess = InferenceSession(os.path.join(test_data_dir, "openai_chat.onnx"),
-                                    self.__opt, providers=["CPUExecutionProvider"])
+            sess = InferenceSession(os.path.join(test_data_dir, "azure_chat.onnx"),
+                                    self.__opt, providers=["CPUExecutionProvider", "AzureExecutionProvider"])
             auth_token = np.array([os.getenv('CHAT', '')])
-            chat = np.array(['{\"model\": \"gpt-3.5-turbo\",\"messages\": [{\"role\": \"system\", \"content\": \"You are a helpful assistant.\"}, {\"role\": \"user\", \"content\": \"Hello!\"}]}'])
+            chat = np.array(['{\"messages\":[{\"role\": \"system\", \"content\": \"You are a helpful assistant.\"},{\"role\": \"user\", \"content\": \"Does Azure OpenAI support customer managed keys?\"},{\"role\": \"assistant\", \"content\": \"Yes, customer managed keys are supported by Azure OpenAI.\"},{\"role\": \"user\", \"content\": \"Do other Azure AI services support this too?\"}]}'])
```
Comment (wenbingl) [inline test/test_azure_ops.py:126]: nit: consider use a Python raw string?

File: test/data/azure/create_basic_models.py (hunk 4 above)
Comment (RandySheriffH) [reply test/data/azure/create_basic_models.py:44]: it temporary, only serving UT request for 1.16 release.

File: test/test_azure_ops.py (hunk 5 above)
Comment (RandySheriffH) [reply test/test_azure_ops.py:126]: Good point.

**Review by wenbingl — APPROVED** [review]

**Pull request merged**